RakeshRenzous/space-rockets
# fix: timezone issue in launch detail page

#### Description
Fixed the timezone issue in the launch detail page to render t in the local timezone of the launch site instead of user local time.

#### Test steps

- Open app, there would be two menu options.
- Click on "Browse SpaceX Launches".
- Click on any Launch item card.
- Should see the timezone of the launch site and on hover should see the user timezone with the respective offset.

#### Thought process 

Saw that the util **formatDateTime** was using _Intl.DateTimeFormat_ and the constructor would take a timezone key and then convert it into the respective timezone, so went through the API to see if the timezone data was provided in the launch API and launch pads API, But neither API had the required data.

So the next thing I tried to search whether if it is possible to convert it into the target timezone by parsing the timeStamp string but it would require additional libraries to do that and I didn't want to add another library and increase the bundle size just for this fix. 

So given that timestamp format would remain the same, I had created a constants file with a compact list of major timezones with offset has the key and timezone name has the value. I was able to get the timezone offset by getting substring of the timestamp and also added a fallback case of rendering it in the user timezone.

#### Before
<img width="1440" alt="Screenshot 2021-11-07 at 5 12 31 AM" src="https://user-images.githubusercontent.com/23741529/140627426-e964764b-cfda-4693-a3d3-9ecd6ea78169.png">

#### After

<img width="1440" alt="Screenshot 2021-11-07 at 6 29 11 AM" src="https://user-images.githubusercontent.com/23741529/140628408-a50e75be-d40d-40d1-bc9c-3c53afcf6bb3.png">
